### PR TITLE
Improve Matches: Part 4

### PR DESCRIPTION
**Pull request by engineer124**

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c [hunk 1]
```diff
@@ -2890,68 +2887,61 @@ void BossTw_Update(Actor* thisx, PlayState* play) {
...
-    if (this->actionFunc != BossTw_Wait) {
-        this->collider.dim.radius = 45;
-
-        if (this->actionFunc == BossTw_Spin) {
-            this->collider.dim.radius *= 2;
-        }
+    if (this->actionFunc == BossTw_Wait) {
+        return;
+    }
+    this->collider.dim.radius = 45;
```
Comment (fig02) [inline src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c:2893]: id line break here personally

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_Bg_Spot00_Hanebasi/z_bg_spot00_hanebasi.c [hunk 2]
```diff
@@ -137,24 +137,25 @@ void BgSpot00Hanebasi_Destroy(Actor* thisx, PlayState* play) {
...
+            child->destAngle = 0;
+            return;
         }
     }
+
+    if ((this->dyna.actor.shape.rot.x == 0) && !IS_CUTSCENE_LAYER && !LINK_IS_ADULT && !IS_DAY) {
+        this->actionFunc = BgSpot00Hanebasi_DrawbridgeRiseAndFall;
+        this->destAngle = -0x4000;
+        child->destAngle = -0xFE0;
+    }
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Spot00_Hanebasi/z_bg_spot00_hanebasi.c:158]: this is probably subjective but it looks a bit better to me as:
```c
    if ((this->dyna.actor.shape.rot.x != 0) && (CutsceneFlags_Get(play, 0) || (!IS_CUTSCENE_LAYER && IS_DAY))) {
        this->actionFunc = BgSpot00Hanebasi_DrawbridgeRiseAndFall;
        this->destAngle = 0;
        child->destAngle = 0;
    } else if ((this->dyna.actor.shape.rot.x == 0) && !IS_CUTSCENE_LAYER && !LINK_IS_ADULT && !IS_DAY) {
        this->actionFunc = BgSpot00Hanebasi_DrawbridgeRiseAndFall;
        this->destAngle = -0x4000;
        child->destAngle = -0xFE0;
    }
```

**Review by Dragorn421 — COMMENTED** [review]
<details>

engineer does it again

</details>

File: src/overlays/actors/ovl_Arrow_Fire/z_arrow_fire.c [hunk 3]
```diff
@@ -200,50 +200,58 @@ void ArrowFire_Draw(Actor* thisx, PlayState* play2) {
...
-        gSPDisplayList(POLY_XLU_DISP++, sModelDL);
-
-        CLOSE_DISPS(play->state.gfxCtx, "../z_arrow_fire.c", 682);
+    if (arrow == NULL) {
+        return;
+    }
+
+    if (arrow->actor.update == NULL) {
+        return;
+    }
```
[the remark this answers is not in the document]
Comment (fig02) [reply src/overlays/actors/ovl_Arrow_Fire/z_arrow_fire.c:210]: fwiw i agree if this still matches. i kinda assumed it didnt, but surely its fine right?

[the remark this answers is not in the document]
Comment (Dragorn421) [reply src/overlays/actors/ovl_Arrow_Fire/z_arrow_fire.c:210]: yeah it does match (the other suggestion too)

[the remark this answers is not in the document]
Comment (engineer124) [reply src/overlays/actors/ovl_Arrow_Fire/z_arrow_fire.c]: Oh nice, I thought I had tested this but apparently not

**Review by fig02 — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by Dragorn421 — APPROVED** [review]

**Pull request merged**